File: backend/app/schemas/logs.py
from datetime import date, datetime

from pydantic import BaseModel, Field


class LogItemRead(BaseModel):
    id: int
    source_type: str
    action_type: str
    object_type: str
    object_id: int | None = None
    object_label: str | None = None
    summary: str
    actor_name: str
    occurred_at: datetime
    team_id: int | None = None
    team_name: str | None = None
    athlete_id: int | None = None
    athlete_name: str | None = None
    session_id: int | None = None
    session_date: date | None = None
    status: str | None = None
    before_snapshot: dict | None = None
    after_snapshot: dict | None = None
    extra_context: dict | None = None


class LogListRead(BaseModel):
    items: list[LogItemRead] = Field(default_factory=list)
    available_object_types: list[str] = Field(default_factory=list)

File: frontend/src/api/logs.ts
import client from './client'

export interface LogItem {
  id: number
  source_type: string
  action_type: string
  object_type: string
  object_id: number | null
  object_label: string | null
  summary: string
  actor_name: string
  occurred_at: string
  team_id: number | null
  team_name: string | null
  athlete_id: number | null
  athlete_name: string | null
  session_id: number | null
  session_date: string | null
  status: string | null
  before_snapshot: Record<string, unknown> | null
  after_snapshot: Record<string, unknown> | null
  extra_context: Record<string, unknown> | null
}

export interface LogListResponse {
  items: LogItem[]
  available_object_types: string[]
}

export async function fetchLogs(params: {
  date_from?: string
  date_to?: string
  actor_name?: string
  object_type?: string
  limit?: number
}) {
  const { data } = await client.get<LogListResponse>('/logs', { params })
  return data
}
